conference/src/main/java/org/spine/samples/lobby/conference/ConferenceRepository.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine.samples.lobby.conference;
+
+import org.spine3.base.EmailAddress;
+import org.spine3.samples.lobby.common.ConferenceId;
+import org.spine3.samples.lobby.conference.contracts.Conference;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+import static com.google.common.collect.Maps.newHashMap;
+
+/**
+ * @author andrii.loboda
+ */
+public class ConferenceRepository {
+    private final Map<ConferenceId, Conference> store = newHashMap();
+
+    public void store(Conference entity) {
+        store.put(entity.getId(), entity);
+    }
+
+    @Nullable
+    public Conference load(ConferenceId id) {
+        return store.get(id);
+    }
+
+    @Nullable
+    public Conference loadByEmailAndAccessCode(EmailAddress emailAddress, String accessCode) {
+        for (Conference conference : store.values()) {
+            final boolean emailMatches = conference.getOwner()
+                                                   .getEmail()
+                                                   .equals(emailAddress);
+            final boolean accessCodeMatches = conference.getAccessCode()
+                                                        .equals(accessCode);
+            if (accessCodeMatches && emailMatches) {
+                return conference;
+            }
+        }
+        return null;
```

Comment:
Yes, we should apply `fail fast` principle. But there is also a best practice to do not return `null`. Actually, this is the quote again from J. Bloch:

> Avoid return values that demand exceptional processing. Clients will forget to write the specialcase code, leading to bugs. For example, return zero-length arrays or collections rather than nulls.

So, I'm sure you'll agree that the better solution is to apply both these principles.

And we have `package-info.java` files with @ParametersAreNonNullByDefault annotation. Please add such file to this package, too.

The classes generated from .proto files also don't accept nulls in setters, and if the field is `null` return the message default instance in getters. I think it is a great approach to use `Optional`, and they just don't use additional libs in the generated code. The advantage on using `Optional` is that the client have to check the value returned. But in this project we already return default messages instead, and if we want to return `Optional`, we have to change this behaviour everywhere (I don't think there are a lot of such places). I understand that this class is raw, draft code, but we should decide which approach to choose. 

Also, we have some utility methods in `Validate` class from Spine like `isDefault()`, etc, check them out.